controllers/registry.go

```diff
@@ -617,30 +687,72 @@ func (r *DPAReconciler) parseAzureSecret(secret corev1.Secret, secretKey string)
 		}
 		// check for storage key
 		matchedStorageKey := azureStorageKeyRegex.MatchString(line)
-
-		if err != nil {
-			r.Log.Info("Error finding storage key for the supplied Azure credential")
-			return AzureStorageKey, err
-		}
+		matchedSubscriptionId := azureSubscriptionIdRegex.MatchString(line)
+		matchedTenantId := azureTenantIdRegex.MatchString(line)
+		matchedCliendId := azureClientIdRegex.MatchString(line)
+		matchedClientsecret := azureClientSecretRegex.MatchString(line)
+		matchedResourceGroup := azureResourceGroupRegex.MatchString(line)
 
 		if matchedStorageKey {
```

Comment:
Would it make sense to have switch-case here ?

Reply:
Dylan and I, spoke about the same. It might involve assigning bits to every field and then using that in switch case. I need to look more into it. Happy to change this block to use switch

Reply:
Changed to `switch` case.